presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -312,6 +312,11 @@ public void testMapSubscript()
         assertQuery("SELECT map(array['a'], array['aa'])['a']", "SELECT 'aa'");
         assertQuery("SELECT map(array[array[1,1]], array['a'])[array[1,1]]", "SELECT 'a'");
         assertQuery("SELECT map(array[(1,2)], array['a'])[(1,2)]", "SELECT 'a'");
+
+        // Force the last value in the key BlockBuilder to the end of the internal Slice, so that searching for longer key values can be tested.
+        // 'idv_fail' and 'request_fail' has the same hash value.
+        assertQuery("SELECT map(array['0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456', 'idv_fail'], array['aa', 'bb'])['request_fail']", "SELECT NULL");
```

Comment:
@yingsu00 I don't think I understand how this works. Would you clarify? Perhaps, a unit tests would be easier to understand. What is the very long string here for?